driver-core/src/test/java/com/datastax/driver/core/WarningsTest.java

```diff
@@ -50,4 +54,52 @@ public void should_expose_warnings_on_execution_info() {
     List<String> warnings = rs.getExecutionInfo().getWarnings();
     assertThat(warnings).hasSize(1);
   }
+
+  @Test(groups = "short")
+  public void should_execute_query_with_server_side_warnings() {
+    // Set the system property to enable logging of server side warnings
+    final String originalLoggingFlag =
+        System.getProperty(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY, "false");
+    System.setProperty(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY, "true");
```

Comment:
Normally tests that manipulate system properties are executed in isolation, but in this case the impact on potentially concurrent tests is insignificant, so I agree with your approach.